lib/cretonne/src/loop_analysis.rs

```diff
@@ -30,14 +31,33 @@ impl PrimaryEntityData for LoopData {}
 
 impl LoopData {
     /// Creates a `LoopData` object with the loop header and its eventual parent in the loop tree.
-    pub fn new(header: Ebb, parent: Option<Loop>) -> LoopData {
+    fn new(header: Ebb, parent: Option<Loop>) -> LoopData {
         LoopData {
             header: header,
             parent: parent.into(),
         }
     }
 }
 
+/// If an `Ebb` is part of a loop, then we record two things: the id of the loop it's part of
+/// and the last instruction in the `Ebb` pertaining to the loop. If the `Ebb` is part of multiple
+/// loops, then we make sure by splitting the `Ebb` that it is part of at most two loops, one being
+/// the direct child of the other, and the parent loop should contain all the `Ebb` up to its
+/// terminator instruction.
+#[derive(Clone,Debug)]
+enum EbbLoopData {
+    NoLoop(),
+    Loop { loop_id: Loop, last_inst: Inst },
```

Comment:
I think this is better represented as two packed options. Both because the enum makes the data structure 50% larger, and because you naturally have three possibilities:

- None, none. No loop
- Loop, None. Whole EBB belongs to a single loop.
- Loop, last. EBB belongs to two loops.